src/element/textWysiwyg.tsx

```diff
@@ -136,25 +139,74 @@ export function textWysiwyg({
   }
 
   function cleanup() {
+    if (isDestroyed) {
+      return;
+    }
+    isDestroyed = true;
     // remove events to ensure they don't late-fire
+    editable.onblur = null;
     editable.onpaste = null;
     editable.oninput = null;
     editable.onkeydown = null;
 
     window.removeEventListener("wheel", stopEvent, true);
+    window.removeEventListener("pointerdown", onPointerDown);
+    window.removeEventListener("pointerup", rebindBlur);
+    window.removeEventListener("blur", handleSubmit);
+
+    unbindUpdate();
+
     document.body.removeChild(editable);
   }
 
+  const rebindBlur = () => {
+    window.removeEventListener("pointerup", rebindBlur);
+    // deferred to guard against focus traps on various UIs that steal focus
+    //  upon pointerUp
+    setTimeout(() => {
+      editable.onblur = handleSubmit;
+      // case: clicking on the same property → no change → no update → no focus
+      editable.focus();
+    });
+  };
+
+  // prevent blur when changing properties from the menu
+  const onPointerDown = (event: MouseEvent) => {
+    if (
+      event.target instanceof HTMLElement &&
+      event.target.closest(".App-menu__left") &&
+      !isWritableElement(event.target)
```

Comment:
A more elegant solution maybe here? We shouldn't treat these class names as permanent identifiers.

Reply:
I'll maybe add the class to `constants.js`?

Reply:
And to guard against regressions, we'll need to test this, which is currently blocked by react-testing-library update.

Reply:
Actually, we should already have a version of `jsdom` that supports selection, but it doesn't work. `npm ls jsdom` returns some weird things so don't know.